### PR TITLE
Use bitcast instead of alloca+store+ptrcast+load sequence.

### PR DESCRIPTION
**Pull request by nlewycky**

### CODE REVIEW

**Comment by nlewycky** [comment]
bors try

File: lib/llvm-backend/src/code.rs [hunk 1]
```diff
@@ -78,22 +78,16 @@ fn trap_if_not_representatable_as_int(
...
 
     let (exponent, invalid_exponent) = {
-        let float_bits = {
-            let space = builder.build_alloca(int_ty, "space");
-            let float_ptr = builder.build_pointer_cast(space, float_ptr_ty, "float_ptr");
-            builder.build_store(float_ptr, value);
-            builder.build_load(space, "float_bits").into_int_value()
-        };
-
+        let float_bits = builder.build_bitcast(value, int_ty, "").into_int_value();
```
Comment (bjfish) [inline lib/llvm-backend/src/code.rs]: Could name be something more descriptive than “” ? Might help with debugging later?

Comment (nlewycky) [reply lib/llvm-backend/src/code.rs]: Done.

**Pull request merged**